### PR TITLE
Fix voice instructions cache

### PR DESCRIPTION
**Pull request by Guardiola31337**
- Fixes `VoiceInstructionLoader` not caching ahead voice instructions

TODOs:

- [x] Clean up code / refactor
  - [x] Remove unnecessary `DEBUG` log calls
  - [x] Get the voice code out of `core` and get all that code out of the processor
    - [x] Listening the milestones from `ui`? Custom progress change listener?
  - [x] Break some logic out to make it testable
    - [x] Add tests
- [x] Remove snapshot repository from root project's `build.gradle` before merging
- [x] Bump MAS version when `4.1.0` gets released 👀 https://github.com/mapbox/mapbox-java/pull/910
- [ ] Implement a better cache management strategy from our side?
  - Right now, caches at the beginning and then on every 5 instructions announced + removing the previous 4 instructions cached. We're requesting voice instructions only once, we should consider increasing API calls though.

### CODE REVIEW

**Comment by Guardiola31337** [comment]
This is updated addressing almost all the `TODO`s from the checklist and ready for a first round of 👀 @danesfeder @devotaaabel

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java [hunk 1]
```diff
@@ -716,6 +716,10 @@ public void toggleHistory(boolean isEnabled) {
     mapboxNavigator.toggleHistory(isEnabled);
   }
 
+  public String retrieveSsmlAnnouncementInstruction(int index) {
```
Comment (danesfeder) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Let's add some javadoc for this new API

**Comment by Guardiola31337** [comment]
@danesfeder this is updated and ready for another round of 👀

**Comment by Guardiola31337** [comment]
Going ahead and merging here as [MAS `4.1.0`](https://github.com/mapbox/mapbox-java/releases/tag/v4.1.0) was released 🎉 

Thanks for the great feedback @danesfeder

**Pull request merged**